app/uk/gov/hmrc/fileupload/ApplicationLoader.scala

```diff
@@ -17,83 +17,112 @@
+import uk.gov.hmrc.play.health.AdminController
 import uk.gov.hmrc.play.http.logging.filters.LoggingFilter
-import uk.gov.hmrc.play.microservice.bootstrap.DefaultMicroserviceGlobal
+import uk.gov.hmrc.play.microservice.bootstrap.Routing.RemovingOfTrailingSlashes
+import uk.gov.hmrc.play.microservice.bootstrap.{JsonErrorHandling, MicroserviceFilters}
 
-import scala.concurrent.{ExecutionContext, Future}
 
-object ControllerConfiguration extends ControllerConfig {
-  lazy val controllerConfigs = Play.current.configuration.underlying.as[Config]("controllers")
-}
-
-object AuthParamsControllerConfiguration extends AuthParamsControllerConfig {
-  lazy val controllerConfigs = ControllerConfiguration.controllerConfigs
-}
-
-object MicroserviceAuditFilter extends AuditFilter with AppName {
-  override val auditConnector = MicroserviceAuditConnector
-
-  override def controllerNeedsAuditing(controllerName: String) = ControllerConfiguration.paramsForController(controllerName).needsAuditing
-}
-
-object MicroserviceLoggingFilter extends LoggingFilter {
-  override def controllerNeedsLogging(controllerName: String) = ControllerConfiguration.paramsForController(controllerName).needsLogging
-}
-
-object MicroserviceAuthFilter extends AuthorisationFilter {
-  override lazy val authParamsConfig = AuthParamsControllerConfiguration
-  override lazy val authConnector = MicroserviceAuthConnector
-
-  override def controllerNeedsAuth(controllerName: String): Boolean = ControllerConfiguration.paramsForController(controllerName).needsAuth
+class ApplicationLoader extends play.api.ApplicationLoader {
+  def load(context: Context) = {
+    LoggerConfigurator(context.environment.classLoader).foreach {
+      _.configure(context.environment)
+    }
+    new ApplicationModule(context).application
+  }
 }
 
-object MicroserviceGlobal extends DefaultMicroserviceGlobal with RunMode {
+class ApplicationModule(context: Context) extends BuiltInComponentsFromContext(context)
+  with AhcWSComponents with AppName with MicroserviceFilters
+  with GraphiteConfig
+  with RemovingOfTrailingSlashes
+  with JsonErrorHandling
+  with ErrorAuditingSettings {
 
-  override val auditConnector = MicroserviceAuditConnector
-
-  override def microserviceMetricsConfig(implicit app: Application): Option[Configuration] = app.configuration.getConfig(s"microservice.metrics")
+  implicit val reader = new UnitOfWorkReader(EventSerializer.toEventData)
+  implicit val writer = new UnitOfWorkWriter(EventSerializer.fromEventData)
 
-  override val loggingFilter = MicroserviceLoggingFilter
+  val subscribe: (ActorRef, Class[_]) => Boolean = actorSystem.eventStream.subscribe
+  val publish: (AnyRef) => Unit = actorSystem.eventStream.publish
+  val withBasicAuth: BasicAuth = BasicAuth(basicAuthConfiguration(configuration))
+  val eventStore = if(play.Environment.simple().isProd && configuration.getBoolean("Prod.mongodb.replicaSetInUse").getOrElse(true)) {
```

Comment:
the previous code used play.Play.isProd which will work in 2.3 at least, play.Environment.simple().isProd is not an equivalent operation as Environment.simple() is defined as below, as you can see it's not the running environment but actually a hard coded value of Mode.TEST

/**
 * A simple environment.
 *
 * Uses the same classloader that the environment classloader is defined in,
 * the current working directory as the path and test mode.
 *
 * @return the environment
 */
public static Environment simple() {
    return new Environment(new File("."), Environment.class.getClassLoader(), Mode.TEST);
}